edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/identity/IIdentity.cs

```diff
@@ -1,10 +1,14 @@
 // Copyright (c) Microsoft. All rights reserved.
 namespace Microsoft.Azure.Devices.Edge.Hub.Core.Identity
 {
+    using Microsoft.Azure.Devices.Edge.Util;
+
     public interface IIdentity
     {
         string Id { get; }
 
-        string IotHubHostName { get; }
+        string IotHubHostname { get; }
+
+        Option<string> GatewayHostname { get; }
```

Comment:
I actually feel we can take this out of the Identity. I know IotHubHostname is here, and this is a similar pattern. But maybe we can take the opportunity to simplify. 
I don't think we need to move IotHubHostname out, but we can just keep GatewayHostname at a global config value.

Reply:
From data modeling point of view, I think it is more appropriate to have gateway hostname inside identity instead of a global config value.  I think adding gateway hostname is still simple and straight-forward.  What is your concern?